packages/backend/src/modules/tvl2/repositories/AmountConfigurationRepository.ts

```diff
@@ -0,0 +1,141 @@
+import { Logger } from '@l2beat/backend-tools'
+// import { Database } from '../../../peripherals/database/Database'
+import { EthereumAddress, ProjectId, UnixTime } from '@l2beat/shared-pure'
+
+import {
+  BaseRepository,
+  CheckConvention,
+} from '../../../peripherals/database/BaseRepository'
+import { Database } from '../../../peripherals/database/Database'
+
+export interface AmountConfigurationRow {
+  id: number
+  indexer_id: string
+  project_id: string
+  chain: string
+  address: string
+  escrow_address: string | null
+  origin: string
+  type: string
+  include_in_total: boolean
+  since_timestamp_inclusive: Date
+  until_timestamp_exclusive: Date | null
+}
+
+export interface AmountConfigurationRecord {
+  id: number
+  indexerId: string
+  projectId: ProjectId
+  chain: string
+  address: EthereumAddress | 'native'
+  escrowAddress?: EthereumAddress
+  origin: 'canonical' | 'external' | 'native'
+  type: 'totalSupply' | 'circulatingSupply' | 'escrow'
+  includeInTotal: boolean
+  sinceTimestampInclusive: UnixTime
+  untilTimestampExclusive?: UnixTime
+}
+
+export class AmountConfigurationRepository extends BaseRepository {
```

Comment:
I think it might be good to actually have this be in AmountRepository. They are very tightly connected and if you are reading one you are likely reading the other.